### DISCUSSION TITLE
Sending volume message using MIDO-MIDI

### DISCUSSION THREAD

**Discussion by rolfiepolfie** [mido/mido#491] [Q&A] [answered]
Sorry for this simple question, the problem may be located  elsewhere like my ports or units, 
could some you make a fast assertion if my messages looks ok.

I want silence and therefore trying to set the volume on my device on channel `16` (channel `15` in code) to zero, but it won't respond. 

Trying both channel volume and expression volume, but not quite sure what is preferred, however I am using one of many channels on my unit so guessing `cc=7` is the one to use when "initiating" a midi-channel, no one of these messages below work, but it react on midi notes.
(maybe my unit must be in a gm/gs mode?)

thanks!!

```python
mido.Message('control_change', channel=15, control=11, value=0)     #expression volume
mido.Message('control_change', channel=15, control=7, value=0)    #channel volume
```

**Comment by rdoursenaud** [comment, marked as answer]
Hi,

Your second message (Channel volume) looks good to me.
It usually is the way to go with most devices.

Could you please tell us which device you're using?
Maybe it does not recognize these control changes.

Also, you're not showing how you're sending the message.
Please extend your code snippet a bit.

Thanks!

P.S.: I have edited your message slightly to add syntax highlighting.

**Reply by rolfiepolfie** [reply]
I must admit a complete downfall of stupidity, when you mentioned sending I realized I haven't send it, must be the Object Orienting programming style working in the more abstract layers for too long and forgetting about the details.
    
```python
message=m.Message('control_change', channel=self._midiChannel_out, control=7, value=vol)        
self._send(message)  
```

**Reply by rdoursenaud** [reply]
No problem. It happens to all of us ;) Glad you fixed your issue!
